mne_connectivity/tests/test_wsmi.py

```diff
@@ -0,0 +1,1065 @@
+    data = rng.randn(n_epochs, n_channels, n_times)
+
+    ch_names = ["Fz", "Cz", "Pz"]
+    info = create_info(ch_names, sfreq=sfreq, ch_types="eeg")
+    epochs = EpochsArray(data, info, tmin=0.0)
+
+    # Set a standard montage
+    montage = mne.channels.make_standard_montage("standard_1020")
+    epochs.set_montage(montage, match_case=False)
+
+    # Should handle small samples gracefully with very low filter frequency
+    result = wsmi(epochs, kernel=2, tau=1)
+
+    # Check the result
+    data_matrix = result.get_data()
+    n_connections = n_channels * (n_channels - 1) // 2
+    assert data_matrix.shape == (n_epochs, n_connections)
+
+    # Test that insufficient samples after time masking raises error
+    with pytest.raises(ValueError, match=r"but at least[\s\S]*are needed"):
+        wsmi(epochs, kernel=5, tau=3, tmin=0.1, tmax=0.15)
+
+
+def test_wsmi_deterministic():
+    """Test that results are deterministic given same input."""
+    sfreq = 100.0
+    n_epochs, n_channels, n_times = 2, 3, 150
+
+    # Create identical datasets
+    rng1 = np.random.RandomState(42)
+    rng2 = np.random.RandomState(42)
+    data1 = rng1.randn(n_epochs, n_channels, n_times)
+    data2 = rng2.randn(n_epochs, n_channels, n_times)
+
+    ch_names = ["Fz", "Cz", "Pz"]
+    info = create_info(ch_names, sfreq=sfreq, ch_types="eeg")
+
+    epochs1 = EpochsArray(data1, info, tmin=0.0)
+    epochs2 = EpochsArray(data2, info, tmin=0.0)
+    montage = mne.channels.make_standard_montage("standard_1020")
+    epochs1.set_montage(montage, on_missing="ignore")
+    epochs2.set_montage(montage, on_missing="ignore")
+
+    # Compute wSMI with identical parameters
+    conn1 = wsmi(
+        epochs1,
+        kernel=3,
+        tau=1,
+    )
+    conn2 = wsmi(
+        epochs2,
+        kernel=3,
+        tau=1,
+    )
+
+    # Results should be identical
+    assert_allclose(conn1.get_data(), conn2.get_data())
+
+
+def test_wsmi_single_channel():
+    """Test error with single channel."""
+    sfreq = 100.0
+    data = np.random.RandomState(0).randn(2, 1, 100)  # Single channel
+
+    info = create_info(["Cz"], sfreq=sfreq, ch_types="eeg")
+    epochs = EpochsArray(data, info, tmin=0.0)
+
+    # Should raise error - single channel connectivity is not meaningful
+    with pytest.raises(ValueError, match="At least 2 channels are required"):
+        wsmi(epochs, kernel=3, tau=1)
+
+
+def test_wsmi_meg_data():
+    """Test with MEG data."""
+    sfreq = 100.0
+    n_epochs, n_channels, n_times = 2, 3, 150
+    data = np.random.RandomState(0).randn(n_epochs, n_channels, n_times)
+
+    # Create MEG channels
+    ch_names = ["MEG0111", "MEG0112", "MEG0113"]
+    info = create_info(ch_names, sfreq=sfreq, ch_types="mag")
+    epochs = EpochsArray(data, info, tmin=0.0)
+
+    # Test with MEG data
+    conn = wsmi(epochs, kernel=3, tau=1)
+
+    assert conn.n_nodes == n_channels
+    assert conn.method == "wSMI"
+
+
+def test_wsmi_all_channels_as_bad():
```

Comment:
you're already testing what happens if 1 channel is bad, and since the behavior is apparently to ignore bad channels, we shouldn't bother also testing when 2, 3, 4, or all channels are bad.